tests/dataplane-performance/setup.md

```diff
@@ -0,0 +1,74 @@
+# Dataplane Performance Testing
+
+## Goals
+
+- To capture the average latency of requests being proxied through NGINX using a variety of routing rules, so that we
+  can see if different routing rules provide different results, and so that we can know if any future work has an impact
+  on data plane performance.
+- A route is created and tested for each routing method below:
+  - path based routing
+  - header based routing
+  - query param based routing
+  - method based routing
+
+## Test Environment
+
+ The following cluster will be sufficient:
+
+- A Kubernetes cluster with 3 nodes on GKE
+  - Node: e2-medium (2 vCPU, 4GB memory)
+
+## Setup
+
+1. Create cloud cluster
+2. Deploy CRDs:
+
+   ```bash
+   kubectl apply -f https://github.com/kubernetes-sigs/gateway-api/releases/download/v0.8.1/standard-install.yaml
+   ```
+
+3. Deploy NGF from edge using Helm install:
+
+   ```console
+   helm install my-release oci://ghcr.io/nginxinc/charts/nginx-gateway-fabric  --version 0.0.0-edge \
+      --create-namespace --wait -n nginx-gateway
+   ```
+
+## Tests
+
+1. First create the test resources. The manifests provided will deploy a single application and service, and two
+   HTTPRoutes with matches for path based, header based, query based, and method based routing.
+
+   ```console
+   kubectl apply -f manifests/gateway.yaml
+   kubectl apply -f manifests/coffee.yaml
+   kubectl apply -f manifests/cafe-routes.yaml
+   ```
+
+2. Get the external IP of the nginx-gateway service and add an entry in /etc/hosts for `<GW_IP> cafe.example.com`.
+
+   ```console
+   kubectl get service my-release-nginx-gateway-fabric \
+   --namespace nginx-gateway \
+   --output jsonpath='{.status.loadBalancer.ingress[0].ip}'
+   ```
+
+3. Run tests using `wrk` (change the file output name to version under test). The tests use `wrk` to send requests to
```

Comment:
if we don't fix the client machine with wrk, I don't think this requirement "so that we can know if any future work has an impact on data plane performance." can be satisfied. 

Could we run the test from a VM on Google Cloud? And later test will use the configuration for the machine.

Reply:
Reran the tests on a VM!